src/cpu-kernels/getitem.cpp

```diff
@@ -2119,7 +2119,7 @@ ERROR awkward_unionarray_regular_index(
     size += fromtags[tagsoffset + i];
   }
   if (size > 0) {
-    I current[size];
+    I* current = new I[size];
```

Comment:
This isn't just Windows; allocating a large array on the stack can overrun the stack—it could have simply been that the stack was smaller or more fully filled in the execution on Windows. Sometimes, the three platforms (Linux, MacOS, Windows) aren't telling us about platform-specific differences, they're just moving things that "shouldn't" matter, revealing undefined behavior that "accidentally" works on the other platforms.

Although now that I think about it,

```c
I current[size]
```

for a non-constant `size` isn't legal C++, I think. Unless I'm mistaken, it's legal C99 but not C++ (one of the few instances where C99 can do more than C++, because the C/C++ split happened before 1999). I haven't looked at the test logs; if it was a segfault it's probably the overrunning stack issue, but if it was a compiler error then only Visual Studio is correctly adhering to C++ semantics. I've noticed that Visual Studio is the only platform that catches "function ends without return statement" errors.